src/Umbraco.Web/Editors/UsersController.cs

```diff
@@ -709,22 +704,26 @@ public async Task<HttpResponseMessage> PostUnlockUsers([FromUri]int[] userIds)
             if (userIds.Length <= 0)
                 return Request.CreateResponse(HttpStatusCode.OK);
 
+            var user = await UserManager.FindByIdAsync(userIds[0].ToString());
+
             if (userIds.Length == 1)
             {
-                var unlockResult = await UserManager.SetLockoutEndDateAsync(userIds[0], DateTimeOffset.Now);
+                
+
+                var unlockResult = await UserManager.SetLockoutEndDateAsync(user, DateTimeOffset.Now);
                 if (unlockResult.Succeeded == false)
                 {
                     return Request.CreateValidationErrorResponse(
                         string.Format("Could not unlock for user {0} - error {1}", userIds[0], unlockResult.Errors.First()));
                 }
-                var user = await UserManager.FindByIdAsync(userIds[0]);
+
                 return Request.CreateNotificationSuccessResponse(
                     Services.TextService.Localize("speechBubbles/unlockUserSuccess", new[] { user.Name }));
             }
 
             foreach (var u in userIds)
             {
-                var unlockResult = await UserManager.SetLockoutEndDateAsync(u, DateTimeOffset.Now);
+                var unlockResult = await UserManager.SetLockoutEndDateAsync(user, DateTimeOffset.Now);
```

Comment:
Not sure we can use `user` here.. Think we need to get the user using `u` before.

Reply:
Well spotted. I have written some basic tests and refactor of this controller action